### ISSUE TITLE
reduce_prod EXTREMELY slow

### ISSUE THREAD

**Issue by sjeps** [tensorflow/tensorflow#23901]
I have a custom operation running tf.reduce_sum over an axis of a Rank 5 tensor, doing an NN training.
As soon as i switch to tf.reduce_prod I get a performance drop of about 20x slower training time.

This was somewhat unexpected since reduce_max or other reductions run at the same speed of reduce_sum.

Is this expected behaviour?

Tested on windows & linux, tensorflow versions tested from 1.5 to 1.11

**Comment by ravikyram** [comment]
@sjeps 

Is this still an issue. Please, close this thread if your issue was resolved. Thanks!
